0/a-burke/molecule.py

```diff
@@ -20,37 +20,48 @@ def __init__(self, geom_str, units):
 	
 	def read(self, geom_str):
 		#Read in the file containing the geometry of the molecule in question
-		print("\n" + "MOLECULE INFORMATION" + "\n")
+		#print("\n" + "MOLECULE INFORMATION" + "\n")
 		geom = []
 		self.labels = []
 		self.masses = []
 		self.charges = []
 		lines = geom_str.strip().splitlines()
 		self.natom = int(lines[0])
-		print("Number of atoms: " + str(self.natom) + "\n")
-		print("Units are currently in: " + self.units + "\n")
+		#print("Number of atoms: " + str(self.natom) + "\n")
+		#print("Units are currently in: " + self.units + "\n")
 		for line in lines[2:]:
 			atom, x, y, z = line.split() 
 			self.labels.append(atom) 
 			geom.append([float(x), float(y), float(z)])
 		self.geom = np.array(geom)	
-		print("Atoms: " + str(self.labels)+ "\n")
-		
+		#print("Atoms: " + str(self.labels)+ "\n")
 		#Reading in the masses and charges of the atoms from the .xyz file
 		
 		for x in self.labels:
 			self.masses.append(get_mass(x))
 			self.charges.append(get_charge(x))
 		y = 0
 		for x in self.labels:
-			print("Mass of " + x + " is: " + str(self.masses[y]))
-			print("Charge of " + x + " is: " + str(self.charges[y]))
+			#print("Mass of " + x + " is: " + str(self.masses[y]))
+			#print("Charge of " + x + " is: " + str(self.charges[y]))
 			y += 1
-		print("\n")
-		print("Molecular Geometry in: " + self.units + "\n")
-		print(self.geom)
+		#print("\n")
+		#print("Molecular Geometry in: " + self.units + "\n")
+		#print(self.geom)
+
+		return self.geom, self.natom, self.labels	
 
+	def xyz_string(self):
+		string = ''
+		i=0
+		string += str(self.natom) + '\n' + str(self.units) + '\n'
+		for n in self.labels:
```

Comment:
This loop structure is fine, but if you wanted to be more compact you could do "for i in range(len(self.geom)):" or I believe you could also do "for line in self.geom:" then replace "self.geom[i]" with "line" on line 59.